internal/provider/common.go

```diff
@@ -222,12 +205,12 @@ func GetActualCreateClusterPlatformParams(platform *models.Platform, userManaged
 				return updatePlatformIsExternal(platform), swag.Bool(true), nil
 			}
 		}
-	} else { // *highAvailabilityMode == models.ClusterHighAvailabilityModeNone
+	} else { // swag.StringValue(highAvailabilityMode) == models.ClusterHighAvailabilityModeNone
 		if isPlatformBM(platform) {
 			return nil, nil, common.NewApiError(http.StatusBadRequest, errors.Errorf("Can't set %s platform on single node OpenShift", *platform.Type))
 		}
 
-		if userManagedNetworking != nil && !*userManagedNetworking {
+		if userManagedNetworking != nil && !swag.BoolValue(userManagedNetworking) {
```

Comment:
I think that there is a reason that we use pointer value and not `swag.BoolValue`.
`swag.BoolValue` also take is consideration the nil case, this is changing the logic behind it.

Reply:
so how come all tests are passing?

Reply:
Because that we do check before if the value of `userManagedNetworking` is nil or not. I'm worried more form a future changes.